src/emulator/storage/apis/firebase.ts

```diff
@@ -128,6 +135,32 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
     return res.json(new OutgoingFirebaseMetadata(metadata));
   });
 
+  firebaseStorageAPI.post(`/b/:bucketId/o/:objectId[(:)]generateSignedUrl`, async (req, res) => {
+    let signedUrlResponse: CreateSignedUrlResponse;
+    try {
+      signedUrlResponse = await storageLayer.generateSignedUrl({
+        bucketId: req.params.bucketId,
+        decodedObjectId: decodeURIComponent(req.params.objectId),
+        authorization: req.header("authorization"),
+        baseUrl: `${req.protocol}://${req.hostname}:${req.socket.localPort}`,
+        ttlSeconds: req.body.ttlSeconds ?? SIGNED_URL_DEFAULT_TTL_SECONDS,
```

Comment:
`req.body.ttlSeconds` is not necessarily a Number right? I think we need to validate it. It's worth extracting a helper then calling it to validate ttlSeconds in both the post and get requests.

Reply:
i validate it in storageLayer